unreleased/multi-package.md

```diff
@@ -0,0 +1,462 @@
+
+`multi-package.yaml`
+```yaml
+packages:
+- <path to first package>
+- <path to second package>
+```
+
+The developer then runs `daml build --all` from the root of their project to build all of the packages listed in the `multi-package.yaml` file, or they run `daml build` within a package to build it and its dependencies.
+
+The rest of this document expands on use cases and usage of the CLI and the `multi-package.yaml` file.
+
+## Problem Statement
+
+Frequently when developing Daml, we have the notion of our "project", which contains all of the code related our Daml application. Within this project, we may have multiple Daml packages, each with its own `daml.yaml`.
+
+For example, say a developer is developing an application with some data model, and some logic over that model in a separate package. To make changes, the developer simultaneously works on two packages Logic and Model, where Logic depends on the build artifact of Model, `Model.dar`.
+
+```
+> tree
+.
+├── package-Logic
+│   ├── daml/...
+│   └── daml.yaml
+└── package-Model
+    ├── .daml/dist/package-Model-1.0.0.dar
+    ├── daml/...
+    └── daml.yaml
+```
+
+Without multi-package, if the developer makes changes to the source code of both Logic and Model, the developer must invoke `daml build` from within `./package-Model` to produce `package-Model-1.0.0.dar` before invoking `daml build` within `./package-Logic`. If the developer forgets to compile package Model first, package Logic will build against a stale version `package-Model-1.0.0.dar` that does not reflect the latest changes to Model's source code.
+
+This has a number of drawbacks:
+- The developer must navigate directories and invoke `daml build` twice.
+- The developer must always remember to build in order to propagate changes to package-Logic-1.0.0.dar.
+- If the developer forgets, compilation may succeed without warnings due to the stale package-Model-1.0.0.dar, where it would fail if package-Model-1.0.0.dar were up to date.
+
+#### Correct Usage Without Multi-Package
+
+```bash=
+> # ... make some changes to Model ...
+> cd ./package-Model/  # Navigate to package-Model
+> daml build           # Build Model.dar
+> cd ../package-Logic/ # Navigate to package-Logic
+> daml build           # Build Logic.dar
+```
+
+## Using Multi-Package as a Solution
+
+Using multi-package builds, the developer can register multiple packages as interdependent parts of our "project". Whenever a package A in our project refers to the DAR of another package B in our project, any `daml build` on package A will automatically rebuild B's DAR from source if it has changed via a second invocation of `daml build` (See [Caching](#Caching) for details on what "changed" means here).
+
+The directories containing the `daml.yaml` for the various packages that should be part of our project are listed (preferably with relative paths) under the `packages` header in the `multi-package.yaml`.
+
+### Example
```

Comment:
In the Daml docs, we can think about ways to thread the example throughout while also making it more concise -- users might not need all the details, especially later in the doc

Reply:
Right okay, i would very much appreciate the help on that